src/hu_stuff.c

```diff
@@ -1233,27 +1247,65 @@ static void HU_widget_build_monsec(void)
   secretcolor = (fullsecretcount >= totalsecret) ? '0'+CR_BLUE : '0'+CR_GRAY;
   itemcolor = (fullitemcount >= totalitems) ? '0'+CR_BLUE : '0'+CR_GRAY;
 
+  char kill_str[32], item_str[32], secret_str[32];
+
+  switch ((automapactive && hud_stats_format_map) ? hud_stats_format_map : hud_stats_format)
+  {
+    case STATSFORMAT_RATIO:
```

Comment:
Another remark: This is five groups of three times the same format string. The decision which of these five groups to chose is done three times during each tic. In Crispy, I decided to go with an array of function pointers instead, which would only have to get updated once during `HU_Start()` here.